.github/workflows/engine-build.yml

```diff
@@ -0,0 +1,58 @@
+name: Build Engine
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "36 00 * * *"
```

Comment:
Does it do an early exit (or at least skip most of the steps) if there are no changes compared to yesterday's build so as not to waste minutes?

Reply:
The triggering of the workflow is still not finalized, right now it's on cron just so it rebuild periodically in my fork so I have data on performance, see how cache behavior looks over time (right now it was daily, I've also done testing with 4h) etc.

For the final shape, I want it to be triggered on every push to master, release branches, and for PRs, without any trigger on cron.